pandas/core/groupby/ops.py

```diff
@@ -89,7 +89,7 @@ def __init__(
 
         self._filter_empty_groups = self.compressed = len(groupings) != 1
         self.axis = axis
-        self.groupings = groupings  # type: Sequence[grouper.Grouping]
+        self.groupings = list(groupings)  # type: List[grouper.Grouping]
```

Comment:
Was this necessary? I think Sequence should have been allowable

Reply:
This was an attempt to troubleshoot a (still-failing, mentioned in OP) complaint about a mismatch in the `groupings` annotations.

Reply:
@simonjayhawkins any thoughts on why mypy is still complaining about groupings?

Reply:
I think you're right about the property and the attribute. 

```
diff --git a/pandas/core/groupby/ops.py b/pandas/core/groupby/ops.py
index cbe012012..46a427434 100644
--- a/pandas/core/groupby/ops.py
+++ b/pandas/core/groupby/ops.py
@@ -90,12 +90,16 @@ class BaseGrouper:
 
         self._filter_empty_groups = self.compressed = len(groupings) != 1
         self.axis = axis
-        self.groupings = list(groupings)  # type: List[grouper.Grouping]
+        self._groupings = list(groupings)  # type: List[grouper.Grouping]
         self.sort = sort
         self.group_keys = group_keys
         self.mutated = mutated
         self.indexer = indexer
 
+    @property
+    def groupings(self):
+        return self._groupings
+
     @property
     def shape(self):
         return tuple(ping.ngroups for ping in self.groupings)
```

silences mypy if you're happy with this approach.

Reply:
seems to work, updated